charts/core/src/Chart/Chart.types.ts

```diff
@@ -0,0 +1,65 @@
+import type { XAXisComponentOption, YAXisComponentOption } from 'echarts';
+import type { LineSeriesOption } from 'echarts/charts';
+import type {
+  DatasetComponentOption,
+  GridComponentOption,
+  LegendComponentOption,
+  TitleComponentOption,
+  ToolboxComponentOption,
+  TooltipComponentOption,
+} from 'echarts/components';
+import type { ComposeOption } from 'echarts/core';
+
+import { DarkModeProps, type HTMLElementProps } from '@leafygreen-ui/lib';
+
+type RequiredSeriesProps = 'type' | 'name' | 'data';
+export type SeriesOption = Pick<LineSeriesOption, RequiredSeriesProps> &
+  Partial<Omit<LineSeriesOption, RequiredSeriesProps>>;
+
+/**
+ * TODO: This might need to be improved. `ComposeOption` appears to make most base option
+ * keys "Arrayable". This is making it difficult to properly test partial options on
+ * methods like updateUtils > updateOptions(), since something like `options.grid` could be
+ * an array even if an object.
+ */
+export type ChartOptions = ComposeOption<
+  | TooltipComponentOption
+  | GridComponentOption
+  | DatasetComponentOption
+  | TitleComponentOption
+  | LegendComponentOption
+  | ToolboxComponentOption
+  | XAXisComponentOption
+  | YAXisComponentOption
+> & { series?: Array<SeriesOption> };
```

Comment:
Essentially, I think this might need to be made more granular or custom depending on specific what we want to support. But this is essentially how they recommend producing an options object type